core/backend/src/BackendHubAccess.ts

```diff
@@ -59,28 +72,37 @@ export interface ChangesetFileProps extends ChangesetProps {
 }
 
 /**
- * Properties that specify a range of changesetIds. There are two ways to specify the start of the range. You may either supply:
- * - `first` the ChangeSetId of the first changeset to be returned, or
- * - `after` the *parent* ChangeSetId of the first changeset to be returned.
- *
- * `end` specifies the ChangesetId of the last changeset to be returned. If undefined, all later changesets are returned.
+ * A range of changesets
  * @internal
  */
-export type ChangesetRange =
-  { first: ChangesetId, after?: never, end?: ChangesetId } |
-  { after: ChangesetId, first?: never, end?: ChangesetId };
+export interface ChangesetRange {
+  /** index of the first changeset */
+  first: ChangesetIndex;
+  /** index of last changeset. If undefined, all changesets after first are returned. */
+  end?: ChangesetIndex;
+}
+
+/** The scope of a lock.
+ * @public
+ */
+export enum LockScope {

```

Comment:
Do you want to add something to the NextVersion.md about the change to LockScope and LockLevel api changes or do you have other changes coming?

Reply:
this is all internal for now (even though that enum is marked @public). I want to work on my next PR which will be to rework concurrencyControl and locking in general.

Reply:
sounds good.